ignite/metrics/ssim.py

```diff
@@ -74,23 +75,25 @@ def __init__(
         if any(y <= 0 for y in self.sigma):
             raise ValueError("Expected sigma to have positive number. Got {}.".format(sigma))
 
+        super(SSIM, self).__init__(output_transform=output_transform, device=device)
         self.gaussian = gaussian
         self.c1 = (k1 * data_range) ** 2
         self.c2 = (k2 * data_range) ** 2
         self.pad_h = (self.kernel_size[0] - 1) // 2
         self.pad_w = (self.kernel_size[1] - 1) // 2
         self._kernel = self._gaussian_or_uniform_kernel(kernel_size=self.kernel_size, sigma=self.sigma)
-        super(SSIM, self).__init__(output_transform=output_transform)
 
     @reinit__is_reduced
     def reset(self) -> None:
-        self._sum_of_batchwise_ssim = 0.0
+        self._sum_of_batchwise_ssim = 0.0  # Not a tensor because batch size is not known in advance.
         self._num_examples = 0
         self._kernel = self._gaussian_or_uniform_kernel(kernel_size=self.kernel_size, sigma=self.sigma)
 
     def _uniform(self, kernel_size):
         max, min = 2.5, -2.5
-        kernel = torch.arange(start=(1 - kernel_size) / 2, end=(1 + kernel_size) / 2, step=1, dtype=torch.float32)
+        kernel = torch.arange(
```

Comment:
Could you please improve the code here and for gaussian kernel:
- replace `torch.arange` by `torch.linspace`

Reply:
Good question if we should put kernel on `self._device`. My reasoning is that, when we specifiy metric's device it is for storage and not for computing with that. So, i'd say let's kernel be of the same device as update's input. What do you think ?

Reply:
Implemented the improved code!

I agree, the kernel should be on the same device as the update's input. I still think it makes sense to initially create the tensor on the device specified by the user, then move it during the first update step (if needed). It doesn't make a big difference either way, so I'm open to whichever one you prefer.

Reply:
OK, let's keep it like that.